### PR TITLE
Sync with 1.x branch

### PR DESCRIPTION
**Pull request by xiangce**

### CODE REVIEW

File: insights/parsers/uname.py [hunk 1]
```diff
@@ -160,6 +160,12 @@ class Uname(Parser):
       ``version-release``.
     - `rhel_release`: A list of two elements, the major and minor RHEL product
       release numbers.
+
+    Further discussion of uname parsing and evaluation can be found in this
+    article:
+
+    `Using uname content in Insights rules <https://mojo.redhat.com/groups/red-hat-insights/blog/2016/03/23/using-uname-content-in-insights-rules>`_
```
Comment (bfahr) [inline insights/parsers/uname.py]: @xiangce we probably should not be referring articles that are not available publicly here.  Would you mind to remove this.

**Comment by bfahr** [comment]
@kylape would you mind to take a quick look at this PR?  Other than my 1 comment it looks good to me.

File: insights/parsers/uname.py (hunk 1 above)
Comment (xiangce) [inline insights/parsers/uname.py]: Sure thing, I will remove it.

File: insights/parsers/ls_boot.py [hunk 2]
```diff
@@ -52,7 +52,8 @@ def parse_content(self, content):
         """
         Parse the directory content.
 
-        One usage example requires a 'data' property for these operations::
```
Comment (xiangce) [inline insights/parsers/ls_boot.py]: About this deprecated ".data", I have replaced it with "files_of('/boot')" in the "missing_boot_files.py". And will create a new PR to remove it.

**Comment by xiangce** [comment]
Since the update of uname (private mojo URL) in this PR is also in `1.x` branch, I have created another PR https://github.com/RedHatInsights/insights-core/pull/92 to remove it from `1.x`.

**Comment by xiangce** [comment]
@kylape - do you have any comments? I would like to merge it after your approve :)

File: insights/core/plugins.py [hunk 3]
```diff
@@ -378,18 +380,18 @@ def register_component(
...
         requires: list of components the component needs
         optional: list of optional components the component can use
-        shared: True for parsers and combiners
+        shared: True for most parsers and shared combiners
         cluster: True for cluster parsers and cluster rules
         emitter: True for components that return make_response(...)
     """
-    is_reducer = component_type not in (parser, cluster_parser)
-    if is_reducer:
+    is_reducers = component_type not in (parser, cluster_parser)
```
Comment (kylape) [inline insights/core/plugins.py]: why did you make this plural?

File: insights/core/plugins.py [hunk 4]
```diff
@@ -276,6 +276,8 @@ def _visit_parsers(parent, visitor):
 
 def get_dependency_graph(component):
     graph = defaultdict(set)
+    if not COMPONENT_DEPENDENCIES[component]:
```
Comment (kylape) [inline insights/core/plugins.py]: Is this code that I had somehow removed?  It's not apparent why this was added in this PR.

File: insights/core/evaluators.py [hunk 5]
```diff
@@ -288,11 +288,11 @@ def run_reducers(self, metadata=None):
             self.parser_results,
             self.all_output,
             self.handle_reducer_error(),
-            stats=self.stats['reducer'])
+            reducer_stats=self.stats['reducer'])
```
Comment (kylape) [inline insights/core/evaluators.py]: Is this is called `reducer_stats` on master as well?

File: insights/core/reducer.py [hunk 6]
```diff
@@ -132,15 +132,16 @@ def _run(reducers, local_output, shared_output, error_handler, output_dict=None,
...
-        r = run_reducer(func, local, shared_output, error_handler, stats=stats)
-        if r:
-            shared_output[func] = r
-            logger.debug("Reducer output: %s", r)
-            if func.shared and output_dict:
-                output_dict[func] = r
-            if func in plugins.EMITTERS:
-                yield func, r
+        log_parser_outputs(func, local, shared_output)
+        if func not in shared_output:
```
Comment (kylape) [inline insights/core/reducer.py]: This doesn't seem related to the name change.

File: setup.cfg [hunk 7]
```diff
@@ -1,6 +1,6 @@
 [tool:pytest]
 # Look for tests only in tests directories.
-python_files = "insights/tests/*" "insights/parsers/tests/*" "insights/combiners/tests/*" "insights/archive/test.py"
+python_files = "insights/tests/*" "insights/parsers/tests/*" "insights/combiners/tests/*" "insights/web/tests/*" "insights/console/tests/*" "insights/archive/test.py"
```
Comment (kylape) [inline setup.cfg]: We don't have `web` or `console` packages in the core.  Is there a reason you added these back?

File: insights/core/reducer.py (hunk 6 above)
Comment (xiangce) [inline insights/core/reducer.py]: Maybe the title is a little misguided, how do you think about to modify the title to Sync with 1.x branch (including the renaming and all other necessary fragmentary)?

Or I create another new PR to sync these trivial details?

**Comment by bfahr** [comment]
@kylape are the changes made by @xiangce satisfactory?

File: insights/core/reducer.py (hunk 6 above)
Comment (kylape) [inline insights/core/reducer.py]: I'd like to see a new PR for this block of changes, just so I can understand why you'd like to port it.

File: insights/core/plugins.py (hunk 4 above)
Comment (kylape) [inline insights/core/plugins.py]: I asked @csams about these changes, and he said they don't need to be ported.  In addition, he has a very large change that will soon be applied to master that will likely make these changes obsolete.

File: insights/core/evaluators.py (hunk 5 above)
Comment (kylape) [inline insights/core/evaluators.py]: I'm not sure, but there's a good chance @csams intentionally renamed this to `stats` since `reducer_stats` doesn't make sense anymore.  I think leaving it as `stats` would be fine, though this might mean changing it on the 1.x branch.

File: insights/core/reducer.py [hunk 8]
```diff
@@ -53,12 +53,12 @@ def collect_results_multi(results_dict):
     return combined_local_output, combined_shared_output
 
 
-def run_multi(parser_output, all_output, error_handler, reducers=plugins.CLUSTER_REDUCERS, stats=None):
+def run_multi(parser_output, all_output, error_handler, reducers=plugins.CLUSTER_REDUCERS, reducer_stats=None, ):
     logger.debug("Multi-node reducer run started")
     if reducers != plugins.CLUSTER_REDUCERS:
-        log_components(reducers)
+        log_reducers(reducers)
```
Comment (kylape) [inline insights/core/reducer.py]: Changing the name to include "reducers" seems to be going in the wrong naming direction.  Perhaps we should change the name on 1.x to `log_components`?

File: insights/parsers/multipath_v4_ll.py [hunk 9]
```diff
@@ -136,7 +136,7 @@ class MultipathDevices(Parser):
         dms (list): Device mapper names of each device, in order found
         aliases (list): Alias of each device, in order found
         wwids (list):  World Wide ID
-        by_dm (dict): Access to each device by device parser name
+        by_dm (dict): Access to each device by device mapper name
```
Comment (kylape) [inline insights/parsers/multipath_v4_ll.py]: This seems to be a regressive name change.

File: insights/tests/__init__.py [hunk 10]
```diff
@@ -160,9 +159,9 @@ def integrate(input_data, module):
     This method drives the mapreduce framework with test `input_data`.
 
     :param InputData input_data: InputData object filled with test data
-    :param module module: module containing parsers and rules to test with
+    :param module module: module containing parsers and reducers to test with
```
Comment (kylape) [inline insights/tests/__init__.py]: All of the changes in this file seem to be regressive name changes.

File: insights/core/plugins.py (hunk 4 above)
Comment (xiangce) [inline insights/core/plugins.py]: OK, will remove them.

File: insights/core/evaluators.py (hunk 5 above)
Comment (xiangce) [inline insights/core/evaluators.py]: OK, will remove this change and will modify the `master` after this PR is merged.

**Comment by xiangce** [comment]
@kylape - Would you please to check it again?

File: insights/parsers/multipath_v4_ll.py (hunk 9 above)
Comment (kylape) [inline insights/parsers/multipath_v4_ll.py]: Ahh, thanks for clarifying.
